examples/dyn_proxy.rs

```diff
@@ -0,0 +1,59 @@
+#![deny(warnings)]
+use warp::hyper::body::{Body, Bytes};
+use warp::hyper::{Client, Request};
+use warp::{
+    http::{method::Method, HeaderMap, Response},
+    path::FullPath,
+    Filter, Rejection,
+};
+
+static PROXY_TARGET: &'static str = "http://httpbin.org";
+
+async fn proxy_request(
+    method: Method,
+    path: FullPath,
+    headers: HeaderMap,
+    body: Bytes,
+) -> Result<Response<Body>, Rejection> {
+    let request = build_request(&method, &path, &headers, body);
+    let client = Client::new();
+    let response = client.request(request).await.unwrap();
+    let response_status = response.status();
+    let response_body = response.into_body();
+
+    Ok(Response::builder()
+        .status(response_status)
+        .body(response_body)
+        .unwrap())
+}
+
+fn build_request(
+    method: &Method,
+    path: &FullPath,
+    headers: &HeaderMap,
+    body: Bytes,
+) -> Request<Body> {
+    let uri = format!("{}/{}", PROXY_TARGET, path.as_str());
+
+    let mut request = Request::builder().method(method.as_str()).uri(uri);
+
+    for (key, value) in headers {
```

Comment:
If you skip using the request builder, you can assign the full headers instead having to iterate here. Like: `*req.headers_mut() = headers;`.

Reply:
Great, i actually tried to do that but couldn't work it out. Thanks, fixed